extension/app/ts/utils/interceptor-messages.ts

```diff
@@ -434,6 +446,83 @@ export const ConfirmTransactionSimulationStateChanged = funtypes.Object({
 	})
 })
 
+export type TabConnection = funtypes.Static<typeof TabConnection>
+export const TabConnection = funtypes.Object({
+	icon: funtypes.String,
+	iconReason: funtypes.String,
+})
+
+export type WebsiteAddressAccess = funtypes.Static<typeof WebsiteAddressAccess>
+export const WebsiteAddressAccess = funtypes.Object({
+	address: EthereumAddress,
+	access: funtypes.Boolean,
+}).asReadonly()
+
+export type LegacyWebsiteAccess = funtypes.Static<typeof WebsiteAccess>
+export const LegacyWebsiteAccess = funtypes.Object({
+	origin: funtypes.String,
+	originIcon: funtypes.Union(funtypes.String, funtypes.Undefined),
+	access: funtypes.Boolean,
+	addressAccess: funtypes.Union(funtypes.ReadonlyArray(WebsiteAddressAccess), funtypes.Undefined),
+})
+export type LegacyWebsiteAccessArray = funtypes.Static<typeof LegacyWebsiteAccessArray>
+export const LegacyWebsiteAccessArray = funtypes.ReadonlyArray(LegacyWebsiteAccess)
+
+export type WebsiteAccess = funtypes.Static<typeof WebsiteAccess>
+export const WebsiteAccess = funtypes.Object({
+	website: Website,
+	access: funtypes.Boolean,
+	addressAccess: funtypes.Union(funtypes.ReadonlyArray(WebsiteAddressAccess), funtypes.Undefined),
+}).asReadonly()
+
+export type WebsiteAccessArray = funtypes.Static<typeof WebsiteAccessArray>
+export const WebsiteAccessArray = funtypes.ReadonlyArray(WebsiteAccess)
+
+export type WebsiteAccessArrayWithLegacy = funtypes.Static<typeof WebsiteAccessArrayWithLegacy>
+export const WebsiteAccessArrayWithLegacy = funtypes.Union(LegacyWebsiteAccessArray, WebsiteAccessArray)
+
+export type UserAddressBook = funtypes.Static<typeof UserAddressBook>
+export const UserAddressBook = funtypes.Object({
+	addressInfos: funtypes.ReadonlyArray(AddressInfo),
+	contacts: ContactEntries,
+})
+
+export type Settings = funtypes.Static<typeof Settings>
+export const Settings = funtypes.Object({
+	activeSimulationAddress: funtypes.Union(EthereumAddress, funtypes.Undefined),
+	activeSigningAddress: funtypes.Union(EthereumAddress, funtypes.Undefined),
+	activeChain: EthereumQuantity,
+	page: Page,
+	makeMeRich: funtypes.Boolean,
+	useSignersAddressAsActiveAddress: funtypes.Boolean,
+	websiteAccess: WebsiteAccessArray,
+	simulationMode: funtypes.Boolean,
+	pendingAccessRequests: PendingAccessRequestArray,
+	userAddressBook: UserAddressBook,
+})
+
+export type UpdateHomePage = funtypes.Static<typeof UpdateHomePage>
+export const UpdateHomePage = funtypes.Object({
+	method: funtypes.Literal('popup_UpdateHomePage'),
+	data: funtypes.Object({
+		simulation: funtypes.Object({
```

Comment:
Should these be readonly?  Same for other objects above.  Unless we specifically need a mutable object, I think we should be using ReadonlyObject and ReadonlyArray everywhere (looks like Array is already Readonly in most places).
```suggestion
export const UpdateHomePage = funtypes.ReadonlyObject({
	method: funtypes.Literal('popup_UpdateHomePage'),
	data: funtypes.ReadonlyObject({
		simulation: funtypes.ReadonlyObject({
```

Reply:
yeah I guess we should have ReadonlyObject everywhere